CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc

```diff
@@ -0,0 +1,245 @@
+// -*- C++ -*-
+//
+// Package:    CondFormats/PCLConfig
+// Class:      AlignPCLThresholdsWriter
+// 
+/**\class AlignPCLThresholdsWriter AlignPCLThresholdsWriter.cc CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
+
+ Description: class to build the SiPixelAli PCL thresholds
+
+*/
+//
+// Original Author:  Marco Musich
+//         Created:  Wed, 22 Feb 2017 12:04:36 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/PCLConfig/interface/AlignPCLThresholds.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+//
+// class declaration
+//
+
+namespace DOFs {
+  enum dof{
+    X,
+    Y,
+    Z,
+    thetaX,
+    thetaY,
+    thetaZ,
+    extraDOF
+  };
+}
+
+class AlignPCLThresholdsWriter : public edm::one::EDAnalyzer<>  {
+   public:
+      explicit AlignPCLThresholdsWriter(const edm::ParameterSet&);
+      ~AlignPCLThresholdsWriter();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+   private:
+      virtual void beginJob() override;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+      virtual void endJob() override;
+      DOFs::dof mapOntoEnum(std::string coord);
+
+      // ----------member data ---------------------------
+      const std::string m_record;
+      const unsigned int m_minNrecords;
+      const std::vector<edm::ParameterSet> m_parameters;
+};
+
+//
+// constructors and destructor
+//
+AlignPCLThresholdsWriter::AlignPCLThresholdsWriter(const edm::ParameterSet& iConfig):
+  m_record(iConfig.getParameter<std::string>("record")),
+  m_minNrecords(iConfig.getParameter<unsigned int>("minNRecords")),
+  m_parameters(iConfig.getParameter<std::vector<edm::ParameterSet> >("thresholds"))
+{
+  //now do what ever initialization is needed
+}
+
+
+AlignPCLThresholdsWriter::~AlignPCLThresholdsWriter()
+{
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+AlignPCLThresholdsWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+   using namespace edm;
+  
+   AlignPCLThresholds* myThresholds = new AlignPCLThresholds();
```

Comment:
Hello @davidlange6 `std::unique_ptr` doesn't work out-of-the-box because of the argument signature of `PoolDBOutputService::writeOne`.
I made the pointer a private member of the class that is then deleted at destruction time. Hope this is fine, let me know.
Marco